### PR TITLE
Charmeditor for forum posts and comments

### PR DESCRIPTION
**Pull request by Devorein**
Removed row actions from the `CharmEditor` for now.
image

### CODE REVIEW

File: components/common/BoardEditor/focalboard/src/components/viewSidebar/viewSourceOptions.tsx [hunk 1]
```diff
@@ -176,6 +176,10 @@ function CharmVerseDatabases(props: DatabaseSourceProps & { activePageId?: strin
           pages={sortedPages}
           activePageId={props.activePageId}
           onSelectPage={onSelect}
+          style={{
+            height: '250px',
```
Comment (mattcasey) [inline components/common/BoardEditor/focalboard/src/components/viewSidebar/viewSourceOptions.tsx:180]: why does a height need to be set here? it should just be max height

Comment (mattcasey) [inline components/common/BoardEditor/focalboard/src/components/viewSidebar/viewSourceOptions.tsx:180]: i think the height should be 100%, why does it need a limit

File: components/common/CharmEditor/CharmEditor.tsx [hunk 2]
```diff
@@ -271,20 +271,33 @@ export function charmEditorPlugins({
...
 
   /** DONT REMOVE THIS STYLING */
   /** ITS TO MAKE SURE THE USER CAN DRAG PAST THE ACTUAL CONTENT FROM RIGHT TO LEFT AND STILL SHOW THE FLOATING MENU */
   left: -50px;
 
+  ${({ colorMode }) =>
+    colorMode === 'dark'
+      ? `
+          background-color: var(--background-light);
+          .ProseMirror[data-placeholder]::before {
```
Comment (mattcasey) [inline components/common/CharmEditor/CharmEditor.tsx:288]: prosemirror should not have to define a background color. that was causing bugs

File: components/common/CharmEditor/CharmEditor.tsx [hunk 3]
```diff
@@ -271,20 +271,33 @@ export function charmEditorPlugins({
...
+      : ''};
+
   /** DONT REMOVE THIS STYLING */
   div.ProseMirror.bangle-editor {
     padding-left: 50px;
     margin-right: -50px;
   }
 
-  code {
+  code {s
```
Comment (mattcasey) [inline components/common/CharmEditor/CharmEditor.tsx:300]: typo

File: components/common/CharmEditor/components/PageList.tsx [hunk 4]
```diff
@@ -11,14 +11,16 @@ interface Props {
...
 }
 
 export default function PagesList({
   emptyText = 'No pages found',
   activeItemIndex = -1,
   activePageId,
   pages,
-  onSelectPage
+  onSelectPage,
+  style
```
Comment (mattcasey) [inline components/common/CharmEditor/components/PageList.tsx:23]: use className instead of style, and styled components please

**Review by mattcasey — APPROVED** [review]
approved, but could be cleaned up a little

File: components/common/BoardEditor/focalboard/src/components/viewSidebar/viewSourceOptions.tsx (hunk 1 above)
Comment (Devorein) [reply components/common/BoardEditor/focalboard/src/components/viewSidebar/viewSourceOptions.tsx:180]: Without a fixed height the inline-database source list makes all the databases visible
image
image

File: components/common/CharmEditor/CharmEditor.tsx (hunk 3 above)
Comment (Devorein) [reply components/common/CharmEditor/CharmEditor.tsx:300]: Thank you 🙏🏼

File: components/common/CharmEditor/components/PageList.tsx (hunk 4 above)
Comment (Devorein) [reply components/common/CharmEditor/components/PageList.tsx:23]: This is just a one-off style used only in `viewSourceOptions`, not sure how to make use of `className` in this case. Perhaps `fixedHeight` prop would add a `class` with appropriate styles?

File: components/forum/components/PostPage/components/PostComment.tsx [hunk 5]
```diff
@@ -179,17 +179,26 @@ export function PostComment({
             left: 10
           }}
         />
-        <Box ml={3}>
+        <Box
+          ml={3}
+          sx={{
+            'div.ProseMirror.bangle-editor': {
```
Comment (Devorein) [inline components/forum/components/PostPage/components/PostComment.tsx:185]: I'm not sure how to use a different padding-left for bangle-editor. @mattcasey you mentioned using classes, do you mean a `bangleEditorclassName` prop passed to `CharmEditor` like `no-padding-left`, that will remove the padding-left?

**Review by mattcasey — APPROVED** [review]

**Pull request merged**